### PR TITLE
Set default resource request value as a fixed fraction of machine capacity on 'GetNonzeroRequest'.

### PR DESCRIPTION
**Pull request by tossmilestone**
**What this PR does / why we need it**:
Fixes `TODO` below:
https://github.com/kubernetes/kubernetes/blob/f9f5677b3edecfc73af3e643db814e716b7f694f/pkg/scheduler/algorithm/priorities/util/non_zero.go#L36-L38

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

The default fraction values could be discussed to be more appropriate.

**Release note**:

```release-note
Set default resource request value as a fixed fraction of machine capacity on 'GetNonzeroRequest'.
```

### CODE REVIEW

**Comment by tossmilestone** [comment]
@davidopp @k82cn please help to review the PR and give your comments, thanks!
:smiley:

**Comment by k82cn** [comment]
sure

**Comment by thockin** [comment]
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/60698)

File: pkg/scheduler/algorithm/priorities/util/non_zero.go [hunk 1]
```diff
@@ -32,20 +32,34 @@ const DefaultMilliCPURequest int64 = 100 // 0.1 core
...
+const DefaultMilliCPURequestFraction = 0.1
+
+// DefaultMemoryRequestFraction defines default memory request fraction.
+const DefaultMemoryRequestFraction = 0.1
+
 // GetNonzeroRequests returns the default resource request if none is found or what is provided on the request
-// TODO: Consider setting default as a fixed fraction of machine capacity (take "capacity v1.ResourceList"
-// as an additional argument here) rather than using constants
-func GetNonzeroRequests(requests *v1.ResourceList) (int64, int64) {
+func GetNonzeroRequests(requests *v1.ResourceList, capacity *v1.ResourceList) (int64, int64) {
```
Comment (k82cn) [inline pkg/scheduler/algorithm/priorities/util/non_zero.go]: honestly, I'm a little confuse about this `TODO`. If we use 0.1 as fraction, there'll be 3.2 cores as default for a 32 cores host, that's a little big/many to me :)

Comment (tossmilestone) [reply pkg/scheduler/algorithm/priorities/util/non_zero.go]: Thanks for your review! I'm not quite sure about the value, so needs some suggestions.
The alternative value is 0.1/32 if hosts have an average 32 cores, that will be consistent with the origin 0.1 core. WDYT @k82cn

Comment (k82cn) [reply pkg/scheduler/algorithm/priorities/util/non_zero.go]: hm... would you help to check who adds this comments and ping him for more suggestion? Honestly, a constant is OK to me for this case.

**Comment by k82cn** [comment]
/unassign @davidopp
/assign

I'll help to review this PR :)

File: pkg/scheduler/algorithm/priorities/util/non_zero.go (hunk 1 above)
Comment (tossmilestone) [reply pkg/scheduler/algorithm/priorities/util/non_zero.go]: ping @hongchaodeng, you're the owner of the `TODO`. Please give suggestion on the default fraction value, thanks!

**Pull request closed without merging**